File: TTdsa-java/Exercise/src/com/aacirq/dp/N46.java
package com.aacirq.dp;

public class N46 {
    public static void main(String[] args) {
        N46 sol = new N46();

        System.out.println(sol.LastRemaining_Solution(5, 3));
    }

    public int LastRemaining_Solution(int n, int m) {
        if (n == 0)
            return -1;
        int dp = 0;
        for (int i = 2; i <= n; i++) {
            dp = (dp + m) % i;
        }
        return dp;
    }
}

File: TTdsa-cpp/Exercise/dp/N46.cpp
//
// Created by renxin on 2021/6/21.
//

#include "ttdsa.h"

class Solution {
public:
    int LastRemaining_Solution(int n, int m) {
        if (n == 0 || m == 0)
            return -1;
        int f = 0;
        for (int i = 2; i <= n; i++) {
            f = (f + m) % i;
        }
        return f;
    }
};

int main() {
    Solution sol;
    // 3
    cout << sol.LastRemaining_Solution(5, 3) << endl;
    return 0;
}